core/src/main/java/org/jdbi/v3/InferredMapperFactory.java

```diff
@@ -13,44 +13,36 @@
  */
 package org.jdbi.v3;
 
-import java.util.List;
+import static org.jdbi.v3.Types.findGenericParameter;
 
-import com.fasterxml.classmate.ResolvedType;
-import com.fasterxml.classmate.TypeResolver;
+import java.lang.reflect.Type;
 
 import org.jdbi.v3.tweak.ResultSetMapper;
 
 class InferredMapperFactory<X> implements ResultSetMapperFactory
 {
-    private static final TypeResolver TR = new TypeResolver();
-    private final Class<?> maps;
-    private final ResultSetMapper<? extends X> mapper;
+    private final Type maps;
+    private final ResultSetMapper<X> mapper;
 
-    InferredMapperFactory(ResultSetMapper<? extends X> mapper)
+    InferredMapperFactory(ResultSetMapper<X> mapper)
     {
+        this.maps = findGenericParameter(mapper.getClass(), ResultSetMapper.class)
```

Comment:
Optional already has this use case covered: `findGenericParameter(type, supertype).orElseThrow(() -> new FooException("error message"))`